walkthroughs/howto-cross-ekscluster-appmesh-with-cloudmap/cluster2.yaml.template

```diff
@@ -0,0 +1,168 @@
+kind: VirtualNode
+metadata:
+  name: front
+  namespace: ${APP_NAMESPACE}
+spec:
+  meshName: ${APP_NAMESPACE}
+  listeners:
+    - portMapping:
+        port: 8080
+        protocol: http
+  serviceDiscovery:
+    cloudMap:
+      namespaceName: ${CLOUDMAP_NAMESPACE}
+      serviceName: front
+  backends:
+    - virtualService:
+        virtualServiceName: colorapp.${CLOUDMAP_NAMESPACE}
+---
+apiVersion: appmesh.k8s.aws/v1beta1
+kind: VirtualNode
+metadata:
+  name: colorapp-blue
+  namespace: ${APP_NAMESPACE}
+spec:
+  meshName: ${APP_NAMESPACE}
+  listeners:
+    - portMapping:
+        port: 8080
+        protocol: http
+  serviceDiscovery:
+    cloudMap:
+      namespaceName: ${CLOUDMAP_NAMESPACE}
+      serviceName: colorapp
+---
+apiVersion: appmesh.k8s.aws/v1beta1
+kind: VirtualNode
+metadata:
+  name: colorapp-red
+  namespace: ${APP_NAMESPACE}
+spec:
+  meshName: ${APP_NAMESPACE}
+  listeners:
+    - portMapping:
+        port: 8080
+        protocol: http
+  serviceDiscovery:
+    cloudMap:
+      namespaceName: ${CLOUDMAP_NAMESPACE}
+      serviceName: colorapp
+---
+apiVersion: appmesh.k8s.aws/v1beta1
+kind: VirtualService
+metadata:
+  name: colorapp.${CLOUDMAP_NAMESPACE}
+  namespace: ${APP_NAMESPACE}
+spec:
+  meshName: ${APP_NAMESPACE}
+  virtualRouter:
+    name: colorapp-router
+    listeners:
+      - portMapping:
+          port: 8080
+          protocol: http
+  routes:
+    - name: color-route
+      http:
+        match:
+          prefix: /
+        action:
+          weightedTargets:
+            - virtualNodeName: colorapp-red
+              weight: 1
+            - virtualNodeName: colorapp-blue
+              weight: 1
+
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: colorapp-blue
+  namespace: ${APP_NAMESPACE}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: colorapp
+      version: blue
+  template:
+    metadata:
+      annotations:
+        appmesh.k8s.aws/mesh: ${APP_NAMESPACE}
+      labels:
+        app: colorapp
+        version: blue
+    spec:
+      containers:
+        - name: colorapp
+          image: ${COLOR_APP_IMAGE}
+          ports:
+            - containerPort: 8080
+          env:
+            - name: "PORT"
+              value: "8080"
+            - name: "COLOR"
+              value: "blue"
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: colorapp-red
+  namespace: ${APP_NAMESPACE}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: colorapp
+      version: red
+  template:
+    metadata:
+      annotations:
+        appmesh.k8s.aws/mesh: ${APP_NAMESPACE}
+      labels:
+        app: colorapp
+        version: red
+    spec:
+      containers:
+        - name: colorapp
+          image: ${COLOR_APP_IMAGE}
+          ports:
+            - containerPort: 8080
+          env:
+            - name: "PORT"
+              value: "8080"
+            - name: "COLOR"
+              value: "red"
+---
+apiVersion: v1
+kind: Service
```

Comment:
you don't need to define a k8s service, virtual-service should be sufficient.

Reply:
agreed. I removed it